syntax = "proto3";

service SystemInfo {
  rpc GetRAMinfo (RAMRequest) returns (RAMResponse) {}

  rpc GetCPUutilization (CPURequest) returns (CPUResponse) {}

  rpc GetDiskUsage (DiskRequest) returns (DiskResponse) {}

  rpc GetNetworkAdapters (NetworkRequest) returns (NetworkResponse) {}

  rpc GetOsType (OsRequest) returns (OsResponse) {}

}

message OsResponse{
	string isVirtualMachine = 1;
}

message OsRequest {}

message NetworkAdapterInfo{
	string friendly_name = 1;
	string Adapter_Desc = 2;
	string ip_address = 3;
	string ip_mask = 4;
	string gateway = 5;
	string dhcp_server = 6;

}

message NetworkResponse{
	repeated NetworkAdapterInfo networks_adapter = 1;
} 

message NetworkRequest{

}

message DiskInfo{
	string path = 1;
	int64 totalsize = 2;
	int64 usedspace = 3;
	int64 freespace = 4;
	int64 usedpercent = 5;
}

message DiskResponse{
	repeated DiskInfo drives = 1;
}

message DiskRequest{}
 

message RAMRequest {
  
}

message RAMResponse {
  int64 totalRAMinGB = 1;
  int64 usedRAMinGB = 2;
  int64 totalRAMinMB = 3;
  int64 usedRAMinMB = 4;
}

message CPURequest{
	
}

message CPUResponse{
	int64 avgCPUusage = 1;
}



